src/api/app/controllers/webui/workflow_runs_controller.rb

```diff
@@ -1,5 +1,11 @@
 class Webui::WorkflowRunsController < Webui::WebuiController
   def index
     @workflow_runs = WorkflowRunPolicy::Scope.new(User.session, WorkflowRun, { token_id: params[:token_id] }).resolve.page(params[:page])
+    @token = Token::Workflow.find(params[:token_id])
+  end
+
+  def show
```

Comment:
Without authorization, anybody access the show page of a workflow run by by guessing a workflow run's ID. This isn't hard at all since it's just an integer.

Reply:
So true, fixing